packages/nuxt/src/app/composables/router.ts

```diff
@@ -136,8 +136,8 @@ export const navigateTo = (to: RouteLocationRaw | undefined | null, options?: Na
   if (isExternal && !options?.external) {
     throw new Error('Navigating to external URL is not allowed by default. Use `navigateTo (url, { external: true })`.')
   }
-  if (isExternal && parseURL(toPath).protocol === 'script:') {
-    throw new Error('Cannot navigate to an URL with script protocol.')
+  if (isExternal && /^(data|javascript|vbscript):$/.test(parseURL(toPath).protocol || '')) {
+    throw new Error('Cannot navigate to an URL with javascript protocol.')
   }
```

Comment:
What about the following?
```suggestion

  const protocol = parseURL(toPath).protocol

  if (isExternal && /^(data|javascript|vbscript):$/.test(protocol || '')) {
    throw new Error(`Cannot navigate to an URL with ${protocol} protocol.`)
  }
```

Reply:
It definitely needs to be hoisted out, but I want to avoid parsing url if we don't need to (e.g. if it's not an external link)

Reply:
Regex or util only could be scoped out...

Reply:
I believe I have already done this

Reply:
Certainly missed the `isExternal` check, makes sense. Although, three possible protocols are checked and only one is provided in the error message, have you seen the template literal in the above snippet? Does it make sense to pass the one satisfying the regex expression?

> Cannot navigate to a URL with **'${protocol}'** protocol.

Reply:
I think we can probably make it more generic - `unsafe protocol`, and maybe include URL in console log.